### PR TITLE
Feature/saas 15348 azure ensure that enable infrastructure encryption for each storage account in azure storage is set to enabled

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -645,6 +645,7 @@ module.exports = {
         'blobSoftDeletionEnabled'       : require(__dirname + '/plugins/azure/storageaccounts/blobSoftDeletionEnabled.js'),
         'storageAccountsTlsVersion'     : require(__dirname + '/plugins/azure/storageaccounts/storageAccountsTlsVersion.js'),
         'storageAccountHasTags'         : require(__dirname + '/plugins/azure/storageaccounts/storageAccountHasTags.js'),
+        'infrastructureEncryptionEnabled': require(__dirname + '/plugins/azure/storageaccounts/infrastructureEncryptionEnabled.js'),
```
Comment (mehakseedat63) [inline exports.js:648]: change plugin name to infrastructureEncryption
```suggestion
        'infrastructureEncryptionEnabled': require(__dirname + '/plugins/azure/storageaccounts/infrastructureEncryptionEnabled.js'),
```

File: plugins/azure/storageaccounts/infrastructureEncryptionEnabled.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Infrastructure Encryption Enabled',
+    category: 'Storage Accounts',
+    domain: 'Storage',
+    description: 'Ensure that Azure Storage Accounts have Infrastructure Encryption Enabled.',
```
Comment (mehakseedat63) [inline plugins/azure/storageaccounts/infrastructureEncryptionEnabled.js:8]:
```suggestion
    description: 'Ensure that Azure Storage Accounts have infrastructure encryption enabled',
```

File: plugins/azure/storageaccounts/infrastructureEncryptionEnabled.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            if (!storageAccount.data.length) {
+                helpers.addResult(results, 0, 'No storage accounts found', location);
+                return rcb();
+            }
+            for (let account of storageAccount.data) {
+
+                if (!account.id) continue;
+
+                if (account.encryption && account.encryption.requireInfrastructureEncryption){
+                    helpers.addResult(results, 0, 'Storage Account have infrastructure encryption enabled', location, account.id);
```
Comment (mehakseedat63) [inline plugins/azure/storageaccounts/infrastructureEncryptionEnabled.js:40]:
```suggestion
                    helpers.addResult(results, 0, 'Storage Account has infrastructure encryption enabled', location, account.id);
```

File: plugins/azure/storageaccounts/infrastructureEncryptionEnabled.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                } else {
+                    helpers.addResult(results, 2, 'Storage Account does have infrastructure encryption enabled', location, account.id);
```
Comment (mehakseedat63) [inline plugins/azure/storageaccounts/infrastructureEncryptionEnabled.js:42]:
```suggestion
                    helpers.addResult(results, 2, 'Storage Account does not have infrastructure encryption enabled', location, account.id);
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

File: plugins/azure/storageaccounts/infrastructureEncryption.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                return rcb();
+            }
+
+            if (!storageAccount.data.length) {
+                helpers.addResult(results, 0, 'No storage accounts found', location);
+                return rcb();
+            }
+            for (let account of storageAccount.data) {
+
+                if (!account.id) continue;
```
Comment (AkhtarAmir) [inline plugins/azure/storageaccounts/infrastructureEncryption.js:37]:
```suggestion
            }

            for (let account of storageAccount.data) {
                if (!account.id) continue;
```

**Pull request merged**